pkg/kv/kvserver/replicate_queue.go

```diff
@@ -591,6 +610,43 @@ func (decommissionPurgatoryError) PurgatoryErrorMarker() {}
 
 var _ PurgatoryError = decommissionPurgatoryError{}
 
+// processOneChangeWithTracing executes processOneChange within a recording
+// span, such that if we encounter an error or processing takes longer than the
+// configured threshold, the recorded traces from the operation will be output
+// to the logs, along with any associated errors.
+func (rq *replicateQueue) processOneChangeWithTracing(
+	ctx context.Context,
+	repl *Replica,
+	canTransferLeaseFrom func(ctx context.Context, repl *Replica) bool,
+	scatter, dryRun bool,
+) (requeue bool, err error) {
+	tCtx, sp := tracing.EnsureChildSpan(ctx, rq.Tracer, "process replica", tracing.WithRecording(tracingpb.RecordingVerbose))
+	processStart := timeutil.Now()
+	defer func() {
+		processDuration := timeutil.Since(processStart)
+		loggingThreshold := rq.logTracesThresholdFunc(rq.store.cfg.Settings, repl)
+		exceededDuration := loggingThreshold > time.Duration(0) && processDuration > loggingThreshold
+		if err != nil {
+			rec := sp.FinishAndGetConfiguredRecording()
+			log.Warningf(ctx, "error processing replica: %v", err)
+			if rec != nil {
```

Comment:
I don't think we want to break log lines across multiple `log.Warning` or `log.Info` calls. This will allow them to be interleaved with other log statements and could make the logs confusing.